templates/secrets_datastore_crypto_key.yaml

```diff
@@ -20,6 +20,10 @@ metadata:
 type: Opaque
 data:
   # Datastore key used to encrypt/decrypt record for the KV store
+  {{- if kindIs "string" .Values.st2.datastore_crypto_key }}
   datastore_crypto_key: {{ .Values.st2.datastore_crypto_key | b64enc }}
+  {{- else }}
+  datastore_crypto_key: {{ .Values.st2.datastore_crypto_key | toJson | b64enc }}
```

Comment:
I wouldn't add an option for users to shoot themselves in their foot.
Datastore crypto key is an artifact that you generate and copy-paste, instead of trying to split it into pieces or re-format with yaml.

Reply:
Because I shot myself in the foot. JSON is valid YAML. If you don't surround the what you copy/paste with single quotes, it will be parsed into a dictionary and then things break.
This change makes the chart more forgiving.

Reply:
Does it also mean https://github.com/StackStorm/stackstorm-ha/blob/3dcd06b772ca5a2f4133a22fa9b382e03b5b422a/values.yaml#L50-L53 had initially wrong instructions?

Reply:
Yes. This change makes those instructions valid.

Reply:
Can we change the Helm values instructions? Similar to what we had for SSH key

```
datastore_crypto_key: |
  {"hmacKey": {"hmacKeyString": "", "size": 256}, "size": 256, "aesKeyString": "", "mode": "CBC"} 
```

Reply:
Sure.
Which part of this change do you dislike? The `kindIs "String"` check, or passing the value through `toJson`?

Reply:
I updated the PR to drop `toJson` here and update the docs in values.yaml instead.

I would still like to understand your intuition - Why/how does serializing the `datastore_crypto_key` create "an option for users to shoot themselves in their foot"?

Reply:
I think it comes from the expectations that the datastore crypto key is an immutable artifact that users just generate and copy-paste, instead of trying to split it into pieces, re-format, or modify. So it's best to keep that piece of data as is to avoid any further confusion or corner cases.